desc/magnetic_fields.py

```diff
@@ -767,13 +785,36 @@ def field_line_integrate(
         relative and absolute tolerances for ode integration
     maxstep : int
         maximum number of steps between different phis
+    solver: diffrax.Solver
+        diffrax Solver object to use in integration,
+        defaults to Tsit5(), a RK45 explicit solver
+    terminating_event_fxn: fxn
+        Function which takes as input the state of the ODE solution at each timestep and
+        outputs a bool which, if True, terminates the solve at that timestep.
+        NOTE: If the solve is terminated early, the output returned is still
+        length(phis), however all values from the step point the fxn evaluated
+        to True and on will be inf
```

Comment:
Would it be better to return NaN instead of Inf for those values? Maybe it doesn't make a difference for plotting.

Reply:
diffrax default is NaN, I could do that yea